### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by MasterJOI**
Add login + registartion by email and oauth2.

### CODE REVIEW

File: backend/core/app/build.gradle [hunk 1]
```diff
@@ -50,5 +50,5 @@ test {
 flyway {
 	url = 'jdbc:postgresql://localhost:5432/mindbridge'
 	user = 'postgres'
-	password = 'admin'
+	password = 'postgres'
```
Comment (kopot-o-d) [inline backend/core/app/build.gradle:53]: change back

File: frontend/src/helpers/validation.helper.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// eslint-disable-next-line max-len
+const emailRegex = /^\w[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~"-]*@((\[?[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}]?)|(([a-zA-Z0-9][a-zA-Z\-0-9]*\.)+[a-zA-Z]+))$/;
+const passwordRegex = /^(?=^.{5,}$)(?=\S+$).+$/;
+const userNameSurnameRegex = /^[a-zA-Z]{2,40}$/;
+const userNicknameRegex = /^[a-zA-Z0-9- ]{2,40}$/;
+
+export const isValidEmail = (str: string): boolean => emailRegex.test(str) && str.length >= 3 && str.length <= 320;
+export const isValidPassword = (str: string): boolean => passwordRegex.test(str) && str.length >= 5;
+export const isValidNickname = (str: string): boolean => userNicknameRegex.test(str);
+export const isValidNameSurname = (str: string): boolean => userNameSurnameRegex.test(str);
```
Comment (GGrify) [inline frontend/src/helpers/validation.helper.ts:20]: Password is also can not be larger than 32 chars
Check validation rules on trello, there are some changes

File: backend/core/app/src/main/resources/db/common/V2__seed_data.sql [hunk 3]
```diff
@@ -1,7 +1,8 @@
 insert into users
 (id, first_name, last_name, email, password, avatar, verified_email)
 values
-('db127d6b-f2f3-4115-b20a-5b5f67c384aa', 'Anayah', 'Lawson', 'lawson@gmail.com', 'anayahLawson1999', null, true),
+('db127d6b-f2f3-4115-b20a-5b5f67c384aa', 'Anayah', 'Lawson', 'lawson@gmail.com', '{bcrypt}$2y$12$SavVMdq1DZxM/kPmh4NYtePG7iKtTVVeVXU8umP/IytzORIbHVOPC', null, true),
+/*anayahLawson1999*/
```
[the remark this answers is not in the document]
Comment (MasterJOI) [reply backend/core/app/src/main/resources/db/common/V2__seed_data.sql:5]: I`m to, maybe then set password hashes to all users?

File: backend/core/app/src/main/java/com/mindbridge/core/domains/user/UserService.java [hunk 4]
```diff
@@ -1,29 +1,54 @@
...
+	public UserService(UserRepository userRepository, PasswordEncoder passwordEncoder) {
+		this.userRepository = userRepository;
+		this.passwordEncoder = new PasswordConfig().passwordEncoder();
+	}
+
+	@Override
+	public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+		User user = userRepository.findByEmail(email)
+				.orElseThrow(() -> new UsernameNotFoundException("User with email : " + email + " not found."));
+		return new UserPrincipal(user);
```
Comment (Cooplix) [inline backend/core/app/src/main/java/com/mindbridge/core/domains/user/UserService.java:40]: Method has name loadUserByUsername, but it searches user by email
Check AuthService.java and JwtFilter.java - different way to use

File: backend/core/app/src/main/resources/db/common/V2__seed_data.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (GGrify) [reply backend/core/app/src/main/resources/db/common/V2__seed_data.sql:5]: seems good, create new migration

**Review by kopot-o-d — APPROVED** [review]

**Pull request merged**